components/script/dom/node.rs

```diff
@@ -1395,6 +1395,28 @@ impl Node {
             .slottable_data
             .manual_slot_assignment = manually_assigned_slot.map(Dom::from_ref);
     }
+
+    /// Gets the parent of this node with the layout tree
```

Comment:
What do you mean by the "layout tree" in this case?

Reply:
The dom as it is seen by `layout_2020`.
That means that assigned slottables are children of their slots (not their parents) and shadow roots are children of their hosts.

I'm not really sure about the terminology here, feel free to suggest something better (:

Reply:
I suppose "layout tree" incorrectly sounds like it refers to the [box tree](https://www.w3.org/TR/css-display-3/#box-tree) from CSS.

Reply:
Is this similar to the "flat tree" concept used by Gecko? It seems reasonable to share terminology. You're right that the conflation with the trees built during layout prompted me to ask the question. Flow tree / render tree / layout trees all essentially suggest those trees.

Reply:
> Is this similar to the "flat tree" concept used by Gecko?

Yes it's the same concept. I've changed to the names to reflect that.